### DISCUSSION TITLE
Broadcast channel pattern question

### DISCUSSION THREAD

**Discussion by maclonghorn** [laravel/framework#51600] [General]
How can we setup a channel pattern where one of the tokens may contain a "dot"?  For example, we use email address for user name, so if we want a channel for a specific user, the channel config would be ($user is authenticated user from JWT token):
```
Broadcast::channel('user.{username}', function ($user, $username) {
    return $user->login_name == $username;
});
```

In our case the channel is "user.joe@blow.com", so `channelNameMatchesPattern` (from /Illuminate/Broadcasting/Broadcasters/Broadcaster.php) returns false.

We just upgraded to Laravel 10 from 8, where it was working fine.

**Comment by henzeb** [comment]
Replace dot with dash?

**Reply by maclonghorn** [reply]
That is one potential solution, but we didn't want to force clients to change.  What we decided instead was to add another channel format to `routes/channels.php`:
```
Broadcast::channel(
    'user.{username}@{domain}.{tld}',
    function ($user, $username, $domain, $tld) {
        return  $user->login_name == "$username@$domain.$tld";
    }
);
```

Having this **in addition to** the other one allows us to handle both username types (email and non-email).
